### PR TITLE
[HighlightCommon] Updating to support transformed objects

### PR DESCRIPTION
**Pull request by LeeHolmes**
This update adds a few changes:

1) If nothing is selected, iterates over all items in the document
2) Filters items to "PartDesign::Body", so that we don't compare against things like origins, planes, and things with no volume
3) Applies global placements to the shapes, so that the intersections go against the transformed objects and not their initial position

Thank you for creating a pull request to contribute to FreeCAD-macros!
To integrate your macro please make sure the following steps are complete:

- [x] Please check this box if you're not submitting a new macro.   
- [ ] Are you submitting a new macro ?  
- [ ] Have you followed the ['How to submit a macro'](../README.md#how-to-submit-a-macro) section of the README.md ?  
- [x] Your macro has a [Description](../README.md#macro-description) in its header.  
- [x] Your macro has a [CamelCase name](../README.md#camelcase-macro-name).  
- [x] Your macro is named [appropriately](../README.md#macro-name-specifics).  
- [x] Your macro contains a [Metadata section](../README.md#macro-metadata) that immediately follows the header description.  
- [x] Your macro is Python3/Qt5 compliant and tested on the latest FreeCAD stable and development releases.  
- [x] You're including documentation on how your macro works (bonus: screenshots and/or video on the Wiki)  
- [x] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)  
- [x] Commit message is titled in the following way `[MacroName] Short description`.
- [ ] Optional, write or update the changelog in the macro, from latest to oldest.

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.19 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=34586).

---

### CODE REVIEW

File: Utility/HighlightCommon.FCMacro [hunk 1]
```diff
@@ -25,52 +25,42 @@ import FreeCADGui as gui
...
     doc = app.activeDocument()
     colli_grp = None  # Group object to group collisions.
     colli_max = 0  # Maximum collision volume.
+
     # Open a transaction in undo pile.
     doc.openTransaction('Seeking collisions')
 
+    objectsToEnumerate = gui.Selection.getSelection()
+    if len(objectsToEnumerate) < 2:
+        objectsToEnumerate = app.ActiveDocument.Objects
```
Comment (galou) [inline Utility/HighlightCommon.FCMacro:38]: `objectsToEnumerate = doc.Objects`

File: Utility/HighlightCommon.FCMacro [hunk 2]
```diff
@@ -25,52 +25,42 @@ import FreeCADGui as gui
...
+
     # Ensure list of unique objects.
     object_list = []
-    for obj in gui.Selection.getSelection():
+    for obj in objectsToEnumerate:
         if obj not in object_list:
-            object_list.append(obj)
+            if obj.isDerivedFrom("PartDesign::Body"):
```
Comment (galou) [inline Utility/HighlightCommon.FCMacro:44]: Please use single-quotes.

**Comment by LeeHolmes** [comment]
Resolved.

**Comment by galou** [comment]
Thanks!

**Pull request merged**